runtime/doc/eval.txt

```diff
@@ -4418,6 +4420,10 @@ jobsend({job}, {data})					{Nvim} *jobsend()*
 			:call jobsend(j, ["abc", "123\n456", ""])
 < 		will send "abc<NL>123<NUL>456<NL>".
 
+		If the job was started with the rpc option this function
+		cannot be used, instead use |rpcnotify()| and |rpcrequest()|
+		to communicate with the job.
```

Comment:
> Just call everything a job.

Well sockets are not jobs. So I'm more in line with your first suggestion. Both jobs and sockets _are_ channels, and (both kind of) channels can operate either in bytes/raw mode or rpc mode. Jobs can furthermore operate in pty mode (special case of bytes mode).

Thus we could have `chan_send()` (for bytes), `chan_close()`, `chan_request()` and `chan_notify()`, (but still `jobstop` as that is job specific.) 
Alternatively we keep `rpcrequest()` and `rpcnotify()`, and have `send()` or `sendbytes()` and `close()`. For some reason I don't like the `chan_` prefix, maybe because the receiver is not any channel, only a `rpc` channel and a `bytes` channel respectively.
In either case we can add `sockopen()` in analogy with `jobopen()`.

But I would merge this PR as is and make another PR/issue for that, as this PR does remove one distinction while not make anything else worse.